sample-push-notifications/sample-push-notifications/ViewController.m

```diff
@@ -74,17 +76,80 @@ - (void)pushDidReceive:(NSNotification *)notification
     [self.tableView reloadData];
 }
 
-- (void)registerForRemoteNotifications {
+#pragma mark - Push Notifications
+
+- (void)registerForRemoteNotifications{
     
-    UIUserNotificationSettings *settings =
-    [UIUserNotificationSettings settingsForTypes:(UIUserNotificationTypeSound |
-                                                  UIUserNotificationTypeAlert |
-                                                  UIUserNotificationTypeBadge)
-                                      categories:nil];
-    [[UIApplication sharedApplication] registerUserNotificationSettings:settings];
-    [[UIApplication sharedApplication] registerForRemoteNotifications];
+    if(CHECK_VERSION(@"10.0")) { // iOS 10+
+        UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+        center.delegate = self;
+        [center requestAuthorizationWithOptions:(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge) completionHandler:^(BOOL granted, NSError * _Nullable error){
+            if( !error ){
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [[UIApplication sharedApplication] registerForRemoteNotifications];
+                });
+                [self getNotificationSettings];
+            }
+        }];
+    }
+    else { // < iOS 10
+        UIUserNotificationSettings *settings =
+        [UIUserNotificationSettings settingsForTypes:(UIUserNotificationTypeSound |
+                                                      UIUserNotificationTypeAlert |
+                                                      UIUserNotificationTypeBadge)
+                                          categories:nil];
+        [[UIApplication sharedApplication] registerUserNotificationSettings:settings];
+        [[UIApplication sharedApplication] registerForRemoteNotifications];
```

Comment:
Wrong logic. Use the AppDelegate delegate method:
- (void)application:(UIApplication *)application didRegisterUserNotificationSettings:(UIUserNotificationSettings *)notificationSettings  {
    [application registerForRemoteNotifications];
}
iOS 9 and less.